client/src/components/ItineraryList/SelectedLegsComparisonTable.tsx

```diff
@@ -0,0 +1,71 @@
+import { Table } from 'react-bootstrap';
+import { TripPatternType, LegType } from './compareUtils.ts';
+import { formatTime } from '../../util/formatTime.ts';
+import { formatDuration } from '../../util/formatDuration.ts';
+import { formatDistance } from '../../util/formatDistance.ts';
+
+interface SelectedLegsComparisonTableProps {
+  itinerary1: TripPatternType;
+  itinerary2: TripPatternType;
+  selectedLegIds: string[];
+  selectedIndexes: number[];
+  timeZone: string;
+}
+
+export function SelectedLegsComparisonTable({
+  itinerary1,
+  itinerary2,
+  selectedLegIds,
+  selectedIndexes,
+  timeZone,
+}: SelectedLegsComparisonTableProps) {
+  if (selectedLegIds.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className="compare-selected-legs-section">
+      <h6 className="compare-selected-legs-title">Selected Legs Comparison</h6>
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>Itinerary</th>
+            <th>From</th>
+            <th>To</th>
+            <th>Mode</th>
+            <th>Line</th>
+            <th>Start Time</th>
+            <th>End Time</th>
+            <th>Duration</th>
+            <th>Distance</th>
+            <th>Cost</th>
+          </tr>
+        </thead>
+        <tbody>
+          {[itinerary1, itinerary2].map((tripPattern, itineraryIdx) =>
+            tripPattern.legs
+              .filter((leg: LegType) =>
+                selectedLegIds.includes(leg.id || `leg${itineraryIdx + 1}_${tripPattern.legs.indexOf(leg)}`),
+              )
+              .map((leg: LegType, legIdx: number) => (
+                <tr key={`${itineraryIdx}_${legIdx}`}>
```

Comment:
Using iterator indices to create the key is not ideal. Consider what happens when the order changes or the contents change. React could decide not to re-render things because the key didn't change. It is better to create keys from stuff that uniquely identify the rows. For example, legs have IDs in the API. Itinerary could perhaps be identified by something else, like a composite of some of its fields?